x-pack/test/apm_api_integration/tests/services/derived_annotations.ts

```diff
@@ -0,0 +1,181 @@
+    { config: 'basic', archives: [] },
+    () => {
+      describe('when there are multiple service versions', () => {
+        let response: APIReturnType<'GET /api/apm/services/{serviceName}/annotation/search'>;
+
+        before(async () => {
+          const { body: indexExists } = await es.indices.exists({ index: indexName });
+          if (indexExists) {
+            await es.indices.delete({
+              index: indexName,
+            });
+          }
+
+          await es.indices.create({
+            index: indexName,
+            body: {
+              mappings: {
+                properties: {
+                  service: {
+                    properties: {
+                      name: {
+                        type: 'keyword',
+                      },
+                      version: {
+                        type: 'keyword',
+                      },
+                      environment: {
+                        type: 'keyword',
+                      },
+                    },
+                  },
+                  transaction: {
+                    properties: {
+                      type: {
+                        type: 'keyword',
+                      },
+                      duration: {
+                        type: 'long',
+                      },
+                    },
+                  },
+                  observer: {
+                    properties: {
+                      version_major: {
+                        type: 'byte',
+                      },
+                    },
+                  },
+                  processor: {
+                    properties: {
+                      event: {
+                        type: 'keyword',
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          });
+
+          const docs = dates.flatMap((date, index) => {
+            const defaults = {
+              transaction: {
+                type: 'request',
+                duration: 1000000,
+              },
+
+              service: {
+                name: 'opbeans-java',
+                environment: 'production',
+                version: index + 1,
+              },
+              observer: {
+                version_major: 8,
+              },
+              processor: {
+                event: 'transaction',
+              },
+            };
+            return [
+              {
+                ...defaults,
+                '@timestamp': date.toISOString(),
+              },
+              {
+                ...defaults,
+                '@timestamp': new Date(date.getTime() + 30000),
+              },
+              {
+                ...defaults,
+                '@timestamp': new Date(date.getTime() + 60000),
+              },
+            ];
+          });
+
+          await es.bulk({
+            index: indexName,
+            body: docs.flatMap((doc) => [{ index: {} }, doc]),
```

Comment:
Why are you adding an empty doc for every annotation doc?

Reply:
I'm not, I think? This is how the bulk API works: an operation (in this case `index`), followed by the document.

Reply:
Ah yes. i forgot that's how it works. So weird that the ES client doesn't abstract that away if you ask me.